src/utils/GoogleService.ts

```diff
@@ -0,0 +1,66 @@
+import { google } from "googleapis";
+import "dotenv/config";
+import { JWT, GaxiosResponse } from "googleapis-common";
+// import { RequestManager } from "@discordjs/rest";
+// import { sheets } from "googleapis/build/src/apis/sheets";
+
+const page = "BotSheet";
+
+class GoogleService {
+  static async getData(range: string): Promise<GaxiosResponse> {
+    const sheet = google.sheets("v4");
+    const sheetdata = await sheet.spreadsheets.values.get({
+      spreadsheetId: process.env.GOOGLESHEETID,
+      auth: this.auth(),
+      range: page + "!" + range,
+    });
+    return sheetdata;
+  }
+
+  static async linearSearch(key: string, range: string): Promise<number[]> {
```

Comment:
Linear search is not ideal for the requirements of the bot. If possible consider a faster search such as a binary search.